src/platforms/arm/stm32/fastpin_arm_stm32.h

```diff
@@ -51,25 +56,37 @@ template<uint8_t PIN, uint8_t _BIT, uint32_t _MASK, typename _GPIO> class _ARMPI
     inline static port_t hival() __attribute__ ((always_inline)) { return _GPIO::r()->ODR | _MASK; }
     inline static port_t loval() __attribute__ ((always_inline)) { return _GPIO::r()->ODR & ~_MASK; }
     inline static port_ptr_t port() __attribute__ ((always_inline)) { return &_GPIO::r()->ODR; }
+
+#if defined(STM32F2XX)
+    inline static port_ptr_t sport() __attribute__ ((always_inline)) { return &_GPIO::r()->BSRRL; }
+    inline static port_ptr_t cport() __attribute__ ((always_inline)) { return &_GPIO::r()->BSRRH; }
+#else
     inline static port_ptr_t sport() __attribute__ ((always_inline)) { return &_GPIO::r()->BSRR; }
     inline static port_ptr_t cport() __attribute__ ((always_inline)) { return &_GPIO::r()->BRR; }
+#endif
+
     inline static port_t mask() __attribute__ ((always_inline)) { return _MASK; }
 };
 
-#if defined(STM32F10X_MD)
+
 #define _R(T) struct __gen_struct_ ## T
+#define _FL_DEFPIN(PIN, BIT, L) template<> class FastPin<PIN> : public _ARMPIN<PIN, BIT, 1 << BIT, _R(GPIO ## L)> {};
+
+#if defined(STM32F10X_MD)
 #define _RD32(T) struct __gen_struct_ ## T { static __attribute__((always_inline)) inline volatile GPIO_TypeDef * r() { return T; } };
 #define _FL_IO(L,C) _RD32(GPIO ## L);  _FL_DEFINE_PORT3(L, C, _R(GPIO ## L));
+
 #elif defined(__STM32F1__)
-#define _R(T) struct __gen_struct_ ## T
 #define _RD32(T) struct __gen_struct_ ## T { static __attribute__((always_inline)) inline gpio_reg_map* r() { return T->regs; } };
-#define _FL_IO(L,C) _RD32(GPIO ## L); _FL_DEFINE_PORT3(L, C, _R(GPIO ## L));
+#define _FL_IO(L,C) _RD32(GPIO ## L);  _FL_DEFINE_PORT3(L, C, _R(GPIO ## L));
+
+#elif defined(STM32F2XX)
+#define _RD32(T) struct __gen_struct_ ## T { static __attribute__((always_inline)) inline volatile GPIO_TypeDef * r() { return T; } };
+#define _FL_IO(L,C) _RD32(GPIO ## L);
```

Comment:
I get compiler errors if I try to add `_FL_DEFINE_PORT3`, if you have insight I'm happy to try and improve this. Otherwise I'll let it be a mystery for a future dev.

```
In file included from /home/five/Scripts/Arduino/photon/fastled_demo/lib/FastLED/src/FastLED.h:48,
                 from /home/five/Scripts/Arduino/photon/fastled_demo/src/fastled_demo.ino:8:
/home/five/Scripts/Arduino/photon/fastled_demo/lib/FastLED/src/fastpin_arm_stm32.h: In static member function 'static const void* NSFastLED::__FL_PORT_INFO<0>::portAddr()':
/home/five/Scripts/Arduino/photon/fastled_demo/lib/FastLED/src/fastpin.h:269:65: error: lvalue required as unary '&' operand
  269 |  static const void *portAddr() { return (void*)&__t_baseType::r(); } };
      |                                                                 ^
/home/five/Scripts/Arduino/photon/fastled_demo/lib/FastLED/src/fastpin_arm_stm32.h:72:39: note: in expansion of macro '_FL_DEFINE_PORT3'
   72 | #define _FL_IO(L,C) _RD32(GPIO ## L); _FL_DEFINE_PORT3(L, C, _R(GPIO ## L));
      |                                       ^~~~~~~~~~~~~~~~
/home/five/Scripts/Arduino/photon/fastled_demo/lib/FastLED/src/fastpin_arm_stm32.h:85:1: note: in expansion of macro '_FL_IO'
   85 | _FL_IO(A,0);
```